locksmith/src/websub/jobs/keys.ts

```diff
@@ -1,23 +1,75 @@
+import { Op } from 'sequelize'
 import { Key } from '../../graphql/datasource'
-import { Hook } from '../../models'
+import { Hook, ProcessedHookItem } from '../../models'
 import { TOPIC_KEYS } from '../topics'
 
 import { networkMapToFnResult, notifyHook } from '../helpers'
 
+const FIRST_LIMIT = 50
+
+export async function fetchUnprocessedKeys(network: number) {
+  const keySource = new Key(network)
+  let page = 0
+  const items = []
+
+  while (true) {
+    const keys = await keySource.getKeys({
+      first: FIRST_LIMIT,
+      skip: page ? page * FIRST_LIMIT : 0,
+    })
+
+    const keyIds = keys.map((lock: any) => lock.id)
+
+    const processedLocks = await ProcessedHookItem.findAll({
+      where: {
+        objectId: {
+          [Op.in]: keyIds,
+        },
+      },
+      order: [['id', 'DESC']],
+    })
+
+    const onlyUnprocessed = keys.filter(
+      (key: any) => !processedLocks.find((item) => item.objectId === key.id)
+    )
+
+    items.push(onlyUnprocessed)
+
+    if (!onlyUnprocessed.length || onlyUnprocessed.length < keys.length) {
+      break
+    }
+
+    page += 1
+  }
+  return items.flat()
+}
+
+async function getUnprocessedKeys(network: number) {
+  const results = await fetchUnprocessedKeys(network)
+
+  await ProcessedHookItem.bulkCreate(
```

Comment:
Can we move that to _after_ we performed the notification? So that if the job dies for any reason we're not "missing" things.